Add Classes:

KonstantsinYurkevich opened the pull request:
1. BasePages
2. CartPage
3. LogInPage
4. ProductPages
5. ProductsPage

Simple one test to confirm that all methods are working.
Added README.md

KonstantsinYurkevich committed:
Add Classes:
1. BasePages
2. CartPage
3. LogInPage
4. ProductPages
5. ProductsPage

Simple one test to confirm that all methods are working.
Added README.md

dzmitryrak commented on src/test/java/pages/BasePages.java:
```diff
@@ -0,0 +1,12 @@
+public abstract class BasePages {
+    WebDriver driver;
+
+    public BasePages(WebDriver driver){
```
отформатировать весь проект

KonstantsinYurkevich replied:
Fixed

dzmitryrak commented on src/test/java/pages/CartPage.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.util.Locale;
+
+public class CartPage extends BasePages {
```
BasePage а не BasePages

KonstantsinYurkevich replied:
Fixed

dzmitryrak commented on src/test/java/pages/ProductsPage.java:
```diff
@@ -0,0 +1,26 @@
+        driver.findElement(By.xpath(openButton)).click();
+    }
+    public void addToCart(String name) {
+        driver.findElement(By.xpath("//*[contains(text(),'"+name+"')]//ancestor::div/div/button[contains(@class," +
```
думаю что div/div/ лишнее, можно просто //div

KonstantsinYurkevich replied:
Если просто //div то находит все кнопки, а так одну - нужную)

dzmitryrak commented on src/test/java/tests/HomeTaskTest.java:
```diff
@@ -0,0 +1,72 @@
+        String priceOnPage = allPrice.get(3).getText();
+        String nameOfProduct = driver.findElement(By.cssSelector("#item_5_title_link > div")).getText();
+        driver.findElement(By.name("add-to-cart-sauce-labs-fleece-jacket")).click();
+        boolean buttonClicked = driver.findElement(By.xpath("//*[contains(text(),'Sauce Labs Fleece Jacket')]/ancestor" +
```
удали его

KonstantsinYurkevich replied:
deleted

dzmitryrak commented on src/test/java/tests/ProductsTest.java:
```diff
@@ -0,0 +1,28 @@
+        productsPage.addToCart(testProduct);
+        cartPage.open();
+        cartPage.remove(testProduct);
+        cartPage.continueShopping();
```
огонь. А где валидация?

KonstantsinYurkevich replied:
fixed)

KonstantsinYurkevich committed:
Format code project
Rename BasePages to BasePage
Deleted HomeTaskTest

KonstantsinYurkevich committed:
Add wait.until to BasePage and LogInPage

KonstantsinYurkevich committed:
Added waitPageLoad and waitButton to BasePage.java
Added methods:
1.boolean pageOpened to CartPage.java
2.getProductName to CartPage.java
3.getProductName to ProductPages.java
4.boolean pageOpened to ProductPages.java
5.boolean removeButtonIsDisplayed to ProductPages.java
6.boolean pageOpened to ProductsPage.java
7.getProductName to ProductsPage.java
Deleted  wait.until(ExpectedConditions.) from LogInPage.java
Added 3 tests to ProductsTest.java

KonstantsinYurkevich committed:
Add method removeButtonIsDisplayed in ProductsPage
Add test afterAddingProductFromProductsPageButtonRemoveDisplayed

KonstantsinYurkevich committed:
Add methods for BurgerMenu in BasePage.java
Add CheckOutPages.java
Add CheckOutPages in BaseTest.java
Add tests for Burger Menu and Fields On first checkout page
Add method displaying button login in LogInPage.java
Rename open to openProduct method asserts opened product page or not in ProductsPage.java

KonstantsinYurkevich committed:
Add method getting error massage from checkout page in CheckOutPages.java
Add tests for field on first checkout page in CheckOutPageTest.java
